magda-semantic-indexer-framework/src/indexSchema.ts

```diff
@@ -0,0 +1,113 @@
+import { config } from "./config.js";
+
+export type ItemType = "registryRecord" | "storageObject";
+
+export function createSemanticIndexerMapping() {
+    const knnVectorFieldConfig =
+        config.elasticSearch.indices.semanticIndex.knnVectorFieldConfig;
+
+    if (knnVectorFieldConfig.compressionLevel && knnVectorFieldConfig.encoder) {
+        throw new Error("compressionLevel and encoder cannot be used together");
+    }
+
+    return {
+        indexName: config.elasticSearch.indices.semanticIndex.indexName,
```

Comment:
I guess the index mapping / schema will evolve down the line.
how about we use the indexName for purpose of index versioning (unless we have a better option)?
e.g. we assume the current version is 1.
so the generate the indexName should be `${config.elasticSearch.indices.semanticIndex.indexName-1}`
when release a new version that requires mapping changes, we can increase the version number in config so a new index with new mapping version will be created.